TUI: TypeScript build errors in bc.ts after PR #1085 merge

## Description
TUI fails to build with TypeScript errors after PR #1085 (lint fixes) was merged.

## Build Output
```
$ bun run build
src/services/bc.ts(62,5): error TS2322: Type '{ (command: string, options?: SpawnOptionsWithoutStdio | undefined): ChildProcessWithoutNullStreams; ...' is not assignable to type 'SpawnFn'.
  Types of parameters 'options' and 'args' are incompatible.
    Type 'string[]' has no properties in common with type 'SpawnOptionsWithoutStdio'.
src/services/bc.ts(215,5): error TS18047: 'proc.stdout' is possibly 'null'.
src/services/bc.ts(219,5): error TS18047: 'proc.stderr' is possibly 'null'.
```

## Impact
- TUI cannot be built
- Blocks all TUI development
- Production deployment blocked

## Root Cause
PR #1085 lint fixes may have changed type signatures incompatibly.

## Environment
- Build: 53b8825
- Previous working build: 5b1b1dc

## Severity
P1 - Build blocker

--- a/tui/src/services/bc.ts
+++ b/tui/src/services/bc.ts
@@ -58,8 +58,9 @@ type SpawnFn = (
 /**
  * Injectable spawn function - defaults to node's spawn
  * Tests can override this via _setSpawnForTesting()
+ * Cast required because nodeSpawn has complex overloads with optional args
  */
-let spawn: SpawnFn = nodeSpawn;
+let spawn: SpawnFn = nodeSpawn as unknown as SpawnFn;
 
 /**
  * Set a custom spawn function for testing
@@ -212,11 +213,11 @@ export async function execBc(args: string[]): Promise<string> {
       }
     }, 30000);
 
-    proc.stdout.on('data', (data: Buffer) => {
+    proc.stdout?.on('data', (data: Buffer) => {
       stdout += data.toString();
     });
 
-    proc.stderr.on('data', (data: Buffer) => {
+    proc.stderr?.on('data', (data: Buffer) => {
       stderr += data.toString();
     });
 

